src/main/java/org/peergos/blockstore/TypeLimitedBlockstore.java

```diff
@@ -0,0 +1,75 @@
+package org.peergos.blockstore;
+
+import io.ipfs.cid.Cid;
+import io.ipfs.multihash.Multihash;
+import org.peergos.Hash;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class TypeLimitedBlockstore implements Blockstore {
+
+    private static final Logger LOG = Logger.getLogger(TypeLimitedBlockstore.class.getName());
+
+    private final Blockstore blocks;
+    private final Set<Cid.Codec> allowedCodecs;
+
+    public TypeLimitedBlockstore(Blockstore blocks, Set<Cid.Codec> allowedCodecs) {
+        this.blocks = blocks;
+        this.allowedCodecs = allowedCodecs;
+    }
+
+    public CompletableFuture<Boolean> bloomAdd(Cid cid) {
+        if (allowedCodecs.contains(cid.codec)) {
+            return blocks.bloomAdd(cid);
+        }
+        LOG.log(Level.WARNING, "Failed attempt to call bloomAdd with codec: " + cid.codec);
+        return CompletableFuture.completedFuture(false);
+    }
+
+    @Override
+    public CompletableFuture<Boolean> has(Cid cid) {
+        if (allowedCodecs.contains(cid.codec)) {
+            return blocks.has(cid);
+        }
+        LOG.log(Level.WARNING, "Failed attempt to call has with codec: " + cid.codec);
```

Comment:
This and get will be called millions of times via the bitswap. So I wouldn't log it

Reply:
logging removed from get and has